addons/l10n_in/models/account_invoice.py

```diff
@@ -73,6 +74,60 @@ def _onchange_name_warning(self):
             }}
         return super()._onchange_name_warning()
 
+    @api.depends('invoice_line_ids.product_id', 'company_id.l10n_in_hsn_code_digit')
+    def _compute_hsn_code_warning(self):
+
+        def build_warning(record, action_name, message, views, domain=False):
+            return {
+                'message': message,
+                'action_text': _("View %s", action_name),
+                'action': record._get_records_action(name=_("Check %s") % action_name, target='current', views=views, domain=domain or [])
+            }
+
+        indian_invoice = self.filtered(lambda m: m.country_code == 'IN')
+        for move in indian_invoice:
+            product_lines = move.invoice_line_ids.filtered(lambda l: l.display_type == 'product')
+            if any(l.tax_ids for l in product_lines) and not move.company_id.vat:
+                move.l10n_in_hsn_code_warning = {
+                    'invalid_hsn_digit_in_company': build_warning(
```

Comment:
Seems more like something for onboarding or even when you want to send it than when you create the invoice.